### PR TITLE
Use default hatch spacing for energy density calculation

### PR DESCRIPTION
**Pull request by safeerehman**
- Use `MachineConstants.DEFAULT_HATCH_SPACING` for calculating `build_rate` and `energy_density` for single bead simulations
- The default value is `100um`
- Update tests as required. The `build rate` will go down by 1e4 and subsequently `energy density` goes up by the same magnitude for single bead simulations where the default value will be used. 
- Porosity and Microstructure should remain unaffected. 
- This will lead to consistent order of magnitudes for the `build rate` and `energy density` for all three simulation types. 

After Changes -

image

### CODE REVIEW

File: src/ansys/additive/core/parametric_study/parametric_study.py [hunk 1]
```diff
@@ -653,11 +655,11 @@ def generate_porosity_permutations(
             to :obj:`MAX_SLICING_STRIPE_WIDTH <MachineConstants.MAX_SLICING_STRIPE_WIDTH>`.
         min_energy_density : float, default: None
             Minimum energy density (J/m^3) to use for porosity simulations. Parameter combinations
-            with an area energy density below this value are not included. Area energy density is
+            with an energy density below this value are not included. Energy density is
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/parametric_study/parametric_study.py:658]: Good catch!

File: src/ansys/additive/core/parametric_study/parametric_utils.py [hunk 2]
```diff
@@ -39,26 +43,24 @@ def build_rate(
...
 
     Returns
     -------
     float
-        Volumetric build rate is returned if hatch spacing is provided.
-        Otherwise, an area build rate is returned. If input units are m/s and m,
-        the output units are m^3/s or m^2/s.
+        The volumetric build rate is returned. For single bead simulations,
+        the default hatch spacing provided in machine constants is used.
+        If input units are m/s and m, the output units are m^3/s or m^2/s.
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/parametric_study/parametric_utils.py:54]:
```suggestion
        The volumetric build rate is returned. If input units are m/s and m, the output units are m^3/s.
```
This function doesn't take any action based on simulation type. We shouldn't mention single bead simulations here.

File: src/ansys/additive/core/parametric_study/parametric_utils.py [hunk 3]
```diff
@@ -39,26 +43,24 @@ def build_rate(
         Laser scan speed.
     layer_thickness : float
         Powder deposit layer thickness.
-    hatch_spacing : float, default: None
+    hatch_spacing : float
         Distance between hatch scan lines.
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/parametric_study/parametric_utils.py:47]:
```suggestion
    hatch_spacing : float, default :obj:`DEFAULT_HATCH_SPACING <MachineConstants.DEFAULT_HATCH_SPACING>`
        Distance between hatch scan lines. If not specified, the default value is used which is in meters and assumes
        the other values are in meters per second and meters.
```

File: src/ansys/additive/core/parametric_study/parametric_utils.py [hunk 4]
```diff
@@ -74,15 +76,15 @@ def energy_density(
...
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/parametric_study/parametric_utils.py:80]:
```suggestion
    hatch_spacing : float, default :obj:`DEFAULT_HATCH_SPACING <MachineConstants.DEFAULT_HATCH_SPACING>`
        Distance between hatch scan lines. If not specified, the default value is used which is in meters and assumes
        the other values are in meters per second and meters.
```

File: src/ansys/additive/core/parametric_study/parametric_utils.py [hunk 5]
```diff
@@ -74,15 +76,15 @@ def energy_density(
...
 
     Returns
     -------
     float
-        Volumetric energy density is returned if hatch spacing is provided.
-        Otherwise an area energy density is returned. If input units are W, m/s, m, or m,
-        the output units are J/m^3 or J/m^2.
+        The volumetric energy density is returned. For single bead simulations,
+        the default hatch spacing provided in machine constants is used.
+        If input units are m/s and m, the output units are m^3/s or m^2/s.
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/parametric_study/parametric_utils.py:87]:
```suggestion
        The volumetric energy density is returned. Energy density is defined as
        laser power / (scan speed * layer thickness * hatch spacing). 
        If input units are m/s and m, the output units are J/m^3.
```

**Review by pkrull-ansys — APPROVED** [review]
Approved with suggestions. Have Julie review this too.

File: src/ansys/additive/core/parametric_study/parametric_study.py [hunk 6]
```diff
@@ -476,14 +476,16 @@ def generate_single_bead_permutations(
...
-        min_area_energy_density : float, default: None
-            Minimum area energy density (J/m^2) to use for single bead simulations.
-            Parameter combinations with an area energy density below this value are
-            not included. Area energy density is defined as laser power / (layer thickness * scan speed).
-        max_area_energy_density : float, default: None
-            Maximum area energy density (J/m^2) to use for single bead simulations.
-            Parameter combinations with an area energy density above this value are
-            not included. Area energy density is defined as laser power / (layer thickness * scan speed).
+        min_energy_density : float, default: None
+            Minimum energy density (J/m^3) to use for single bead simulations. The default hatch
```
Comment (julieatansys) [inline src/ansys/additive/core/parametric_study/parametric_study.py:480]: For better logical flow, I would reorder these sentences as follows:

Minimum energy density (J/m^3) to use for single bead simulations. Parameter combinations with an energy density below this value are not included. Energy density is defined as laser power / (layer thickness * scan speed * hatch spacing). The default hatch spacing value of :obj:`DEFAULT_HATCH_SPACING <MachineConstants.DEFAULT_HATCH_SPACING>` is used.

File: src/ansys/additive/core/parametric_study/parametric_study.py [hunk 7]
```diff
@@ -476,14 +476,16 @@ def generate_single_bead_permutations(
...
+            spacing value of :obj:`DEFAULT_HATCH_SPACING <MachineConstants.DEFAULT_HATCH_SPACING>` is used.
+            Parameter combinations with an energy density below this value are not included. Energy density is
+            defined as laser power / (layer thickness * scan speed * hatch spacing).
+        max_energy_density : float, default: None
+            Maximum energy density (J/m^3) to use for single bead simulations. The default hatch
```
Comment (julieatansys) [inline src/ansys/additive/core/parametric_study/parametric_study.py:485]: For better logical flow, I would reorder these sentences as follows:

Maximum energy density (J/m^3) to use for single bead simulations. Parameter combinations with an energy density above this value are not included. Energy density is defined as laser power / (layer thickness * scan speed * hatch spacing). The default hatch spacing value of :obj:`DEFAULT_HATCH_SPACING <MachineConstants.DEFAULT_HATCH_SPACING>` is used.

**Comment by pkrull-ansys** [comment]
@safeerehman Don't merge this yet. There is ongoing discussion on whether we want to keep Area Energy Density.

**Comment by safeerehman** [comment]
Closing PR as requirements have changed. Please refer https://tfs.ansys.com:8443/tfs/ANSYS_Development/Portfolio/_workitems/edit/1126126

**Pull request closed without merging**